From af5836e99050ef940d2ded3823b72faee28e3412 Mon Sep 17 00:00:00 2001
From: David Engel <dengel@mythtv.org>
Date: Wed, 18 Jun 2014 14:26:04 -0500
Subject: [PATCH 09/10] Try to prevent duplicate recording rules from getting
 created.

Add a unique index on the record table to prevent the creation of
duplicate recording rules, where duplicate for this means having the
same chanid, starttime/date, title and rule type.  This is mainly
intended for the Web GUI where an accidental double click could result
in truly duplicate rules.
---
 mythtv/bindings/perl/MythTV.pm          |    2 +-
 mythtv/bindings/python/MythTV/static.py |    2 +-
 mythtv/libs/libmythbase/mythversion.h   |    2 +-
 mythtv/libs/libmythtv/dbcheck.cpp       |   12 ++++++++++++
 4 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/mythtv/bindings/perl/MythTV.pm b/mythtv/bindings/perl/MythTV.pm
index e4e03dc..ea02eb8 100644
--- a/mythtv/bindings/perl/MythTV.pm
+++ b/mythtv/bindings/perl/MythTV.pm
@@ -115,7 +115,7 @@ package MythTV;
 # schema version supported in the main code.  We need to check that the schema
 # version in the database is as expected by the bindings, which are expected
 # to be kept in sync with the main code.
-    our $SCHEMA_VERSION = "1327";
+    our $SCHEMA_VERSION = "1328";
 
 # NUMPROGRAMLINES is defined in mythtv/libs/libmythtv/programinfo.h and is
 # the number of items in a ProgramInfo QStringList group used by
diff --git a/mythtv/bindings/python/MythTV/static.py b/mythtv/bindings/python/MythTV/static.py
index b1101c0..384efed 100644
--- a/mythtv/bindings/python/MythTV/static.py
+++ b/mythtv/bindings/python/MythTV/static.py
@@ -5,7 +5,7 @@ Contains any static and global variables for MythTV Python Bindings
 """
 
 OWN_VERSION = (0,28,-1,0)
-SCHEMA_VERSION = 1327
+SCHEMA_VERSION = 1328
 NVSCHEMA_VERSION = 1007
 MUSICSCHEMA_VERSION = 1018
 PROTO_VERSION = '82'
diff --git a/mythtv/libs/libmythbase/mythversion.h b/mythtv/libs/libmythbase/mythversion.h
index eafc0ba..b0aa834 100644
--- a/mythtv/libs/libmythbase/mythversion.h
+++ b/mythtv/libs/libmythbase/mythversion.h
@@ -61,7 +61,7 @@
  *      mythtv/bindings/php/MythBackend.php
 #endif
 
-#define MYTH_DATABASE_VERSION "1327"
+#define MYTH_DATABASE_VERSION "1328"
 
 
  MBASE_PUBLIC  const char *GetMythSourceVersion();
diff --git a/mythtv/libs/libmythtv/dbcheck.cpp b/mythtv/libs/libmythtv/dbcheck.cpp
index 167999d..a885491 100644
--- a/mythtv/libs/libmythtv/dbcheck.cpp
+++ b/mythtv/libs/libmythtv/dbcheck.cpp
@@ -2725,6 +2725,18 @@ NULL
             return false;
     }
 
+    if (dbver == "1327")
+    {
+        const char *updates[] = {
+            "ALTER TABLE record DROP INDEX chanid",
+            "ALTER TABLE record ADD UNIQUE INDEX "
+            " (chanid, starttime, startdate, title, type)",
+            NULL
+        };
+        if (!performActualUpdate(updates, "1328", dbver))
+            return false;
+    }
+
     return true;
 }
 
-- 
1.7.10.2

